test/LiquidityManager/invest.test.ts

```diff
@@ -152,165 +235,123 @@ describe.only('LiquidityManager#invest', () => {
+        const pool = await UniswapV3Helpers.getPoolByContract(stableBtcLpUniV3)
 
-        await liquidityManager
-            .connect(account0)
-            .investUniswapV3(
-                {
-                    positionManager: positionManagerUniV3,
-                    inputToken: stablecoin,
-                    depositAmountInputToken: amount,
-
-                    fee: 3_000,
-
-                    token0,
-                    token1,
-
-                    swapToken0: stableIsToken0 ? '0x' : encodedSwap,
-                    swapToken1: stableIsToken0 ? encodedSwap : '0x',
-
-                    swapAmountToken0: stableIsToken0 ? stableAmount2 : stableAmount1,
-                    swapAmountToken1: stableIsToken0 ? stableAmount1 : stableAmount2,
+        const { token0, token1 } = UniswapV3.sortTokens(
+            await mockTokenWithAddress(USD_PRICE_BN, 18, stablecoin),
+            await mockTokenWithAddress(BTC_PRICE_BN, 18, wbtc),
+        )
 
-                    // TODO calculate tick dinamically using price
-                    tickLower: getNearestUsableTick(tick - (tick / 10n), tickSpacing),
-                    tickUpper: getNearestUsableTick(tick + (tick / 10n), tickSpacing),
+        const {
+            amount0,
+            amount1,
+            tickLower,
+            tickUpper,
+            swapAmountToken0,
+            swapAmountToken1,
+        } = UniswapV3.getMintPositionInfo(
+            amountWithDeductedFees,
+            pool,
+            token0.price,
+            token1.price,
+            10, // 10% lower
+            10, // 10% upper
+        )
 
-                    amount0Min: stableIsToken0 ? stableAmountMin : wbtcAmountMin,
-                    amount1Min: stableIsToken0 ? wbtcAmountMin : stableAmountMin,
-                },
-                permitAccount0,
-            )
+        await investAndValidateTransaction(
+            pool,
+            amount0,
+            amount1,
+            token0,
+            token1,
+            swapAmountToken0,
+            swapAmountToken1,
+            tickLower,
+            tickUpper,
+        )
     })
 
     it('should add liquidity using uniswap v2 and v3 swap in the same transaction', async () => {
-        const halfAmount = amount * 50n / 100n
-        const halfAmountWithDeductedFees = await deductFees(halfAmount)
-
-        const [
-            { tick },
-            tickSpacing,
-            { token0, token1 },
-            wbtcEncodedSwap,
-            wethEncodedSwap,
-        ] = await Promise.all([
-            btcEthLpUniV3.slot0(),
-            btcEthLpUniV3.tickSpacing(),
-            sortTokens(wbtc, weth),
-            getEncodedSwap(halfAmountWithDeductedFees, wbtc, BTC_PRICE_BN),
-            getEncodedSwap(halfAmountWithDeductedFees, weth, ETH_PRICE_BN, 'uniswapV3'),
-        ])
-
-        const wbtcIsToken0 = await isSameToken(wbtc, token0)
+        const pool = await UniswapV3Helpers.getPoolByContract(btcEthLpUniV3)
```

Comment:
this test seems to be the same as the previous test (`should add liquidity and mint a position with expected token amounts`). and doesn't seem to match what the `it` description says it is testing